compiler/rustc_middle/src/ty/generic_args.rs

```diff
@@ -379,6 +379,8 @@ impl<'tcx> GenericArgs<'tcx> {
         self.iter().filter_map(|k| k.as_const())
     }
 
+    /// Returns generic arguments that are not lifetimes. Consider using `non_erasable_generics_real`
+    /// if you have access to the generics declaration that this instance is for.
     #[inline]
     pub fn non_erasable_generics(
```

Comment:
I just did a quick survey, all uses of this function have that information available, even if it requires some minor changes. Instead of passing in the generics, you could pass in the `DefId` to make it less verbose to use.